packages/ui/src/hooks/use-clipboard-otp.ts

```
@@ -0,0 +1,222 @@
+  }, [isSupported, hasPermission, requestPermission, extractOtpFromText]);
+
+  // Listen for permission changes and auto-check clipboard when granted
+  useEffect(() => {
+    if (!isSupported) return;
+
+    let permissionStatus: PermissionStatus | null = null;
+    let handlePermissionChange: (() => void) | null = null;
+
+    const setupPermissionListener = async () => {
+      try {
+        if (navigator.permissions && navigator.permissions.query) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)